plugins/amongus/amongus.cpp

```diff
@@ -5,122 +5,132 @@
-	stream << "Host ID: " << std::to_string(fields.hostId) << '\n';
-	stream << "Public game: " << (fields.isGamePublic ? "true" : "false");
+MumbleStringWrapper mumble_getDescription() {
+	static const char description[] = "Provides positional audio functionality for Among Us."
+									  "Context and identity are provided.";
 
-	identity = utf8ToUtf16(stream.str());
+	MumbleStringWrapper wrapper;
+	wrapper.data           = description;
+	wrapper.size           = sizeof(description) - 1;
+	wrapper.needsReleasing = false;
 
-	return true;
+	return wrapper;
 }
 
-static int tryLock(const std::multimap< std::wstring, unsigned long long int > &pids) {
-	const std::string name = "Among Us.exe";
-	const auto id          = Process::find(name, pids);
-	if (!id) {
-		return false;
-	}
-
-	game = std::make_unique< Game >(id, name);
-	if (!game->isOk()) {
-		game.reset();
-		return false;
-	}
+uint32_t mumble_getFeatures() {
+	return MUMBLE_FEATURE_POSITIONAL;
+}
 
-	const auto fields = game->clientFields();
-	if (!inGame(fields)) {
-		game.reset();
-		return false;
+uint8_t mumble_initPositionalData(const char *const *programNames, const uint64_t *programPIDs, size_t programCount) {
+	for (size_t i = 0; i < programCount; ++i) {
+		if (strcmp(programNames[i], "Among Us.exe") != 0) {
+			continue;
+		}
+
+		game = std::make_unique< Game >(programPIDs[i], programNames[i]);
+
+		const auto error = game->init();
+		if (error == Game::NONE) {
+			const auto fields = game->clientFields();
+			if (!game->isMultiplayer(fields)) {
+				game.reset();
+				return MUMBLE_PDEC_ERROR_TEMP;
+			}
+		} else {
+			game.reset();
+			return error == Game::TEMP ? MUMBLE_PDEC_ERROR_TEMP : MUMBLE_PDEC_ERROR_PERM;
+		}
+
+		return MUMBLE_PDEC_OK;
 	}
 
-	return true;
+	return MUMBLE_PDEC_ERROR_TEMP;
 }
 
-static const std::wstring longDesc() {
-	return std::wstring(L"Supports Among Us with context and identity support.");
+void mumble_shutdownPositionalData() {
+	game.reset();
 }
 
-static std::wstring description(L"Among Us");
-static std::wstring shortName(L"Among Us");
+bool mumble_fetchPositionalData(float *avatarPos, float *avatarDir, float *avatarAxis, float *cameraPos,
+								float *cameraDir, float *cameraAxis, const char **contextPtr,
+								const char **identityPtr) {
+	const auto fields = game->clientFields();
 
-static int tryLock1() {
-	return tryLock(std::multimap< std::wstring, unsigned long long int >());
-}
+	if (!game->isMultiplayer(fields)) {
+		return false;
+	}
 
-static void nullUnlock() {
-}
+	if (fields.gameState == GameState::Ended) {
+		return true;
+	}
 
-static MumblePlugin amongusPlug = { MUMBLE_PLUGIN_MAGIC, description, shortName, nullptr, nullptr, tryLock1,
-									nullUnlock,          longDesc,    fetch };
+	const auto playerControlFields = game->playerControlFields();
 
-static MumblePlugin2 amongusPlug2 = { MUMBLE_PLUGIN_MAGIC_2, MUMBLE_PLUGIN_VERSION, tryLock };
+	// Mumble | Game
+	// X      | X
+	// Y      | -
+	// Z      | Y
```

Comment:
That seems like a very odd mapping to me :eyes: 
Does Mumble not consider the z-axis to be up (as in out of the plane)?

Reply:
Who came up with that weird coordinate system? xD

Reply:
It's "natural" 3D instead of a 2D plane with the third coordinate added to it.

Reply:
Well 3D by definition is a 2D plane with a third coordinate added to it ^^
The only choice is how to orient the plane :shrug:

Anyway - we might want to consider changing Z to being the up-direction at some point as that is what most people probably expect.
But that is of course out of the scope of this PR...

Reply:
For 2D games it would definitely make more sense.